src/main/java/core/basesyntax/dao/jdbc/DaoJdbcCarImpl.java

```diff
@@ -0,0 +1,169 @@
+        String query = "UPDATE cars SET model = ?, manufacturer_id = ? "
+                + "WHERE id = ? AND deleted = false";
+        try (Connection connection = ConnectionUtil.getConnection();
+                    PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setString(1, item.getModel());
+            statement.setLong(2, item.getManufacturer().getId());
+            statement.setLong(3, item.getId());
+            statement.execute();
+            deleteDriversFromCar(item.getId(), connection);
+            addDriversIntoCar(item.getId(), item.getDrivers(), connection);
+        } catch (SQLException throwables) {
+            throw new DataProcessingException("ERROR: can't update car " + item, throwables);
+        }
+        return item;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE cars SET deleted = true WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            return statement.executeUpdate() > 0;
+        } catch (SQLException throwables) {
+            throw new DataProcessingException("ERROR: can't delete car by id " + id, throwables);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String query = "SELECT cars.id, drivers.id, cars.model, cars.manufacturer_id, "
+                + "drivers.deleted FROM cars "
+                + "INNER JOIN cars_drivers cs ON cs.car_id = cars.id "
+                + "INNER JOIN drivers ON cs.driver_id = drivers.id "
+                + "JOIN manufacturers m on cars.manufacturer_id = m.id "
+                + "WHERE m.deleted = false AND cars.deleted = false "
+                + "AND drivers.id = ? ORDER BY cars.id;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, driverId);
+            ResultSet resultSet = statement.executeQuery();
+            List<Car> cars = new ArrayList<>();
+            while (resultSet.next()) {
+                cars.add(DaoUtils.parseCar(resultSet));
+            }
+            return cars;
+        } catch (SQLException throwables) {
+            throw new DataProcessingException("ERROR: can't get all cars by driver with id "
+                    + driverId, throwables);
+        }
+    }
+
+    private List<Driver> getDrivers(Long id, Connection connection) throws SQLException {
+        String query = "SELECT cars.id, drivers.id, drivers.name, drivers.license_number, "
+                + "drivers.deleted FROM cars INNER JOIN cars_drivers cs ON cs.car_id = cars.id "
+                + "    INNER JOIN drivers ON cs.driver_id = drivers.id "
+                + "WHERE cars.id = ? AND drivers.deleted = false ORDER BY drivers.id;";
```

Comment:
```suggestion
        String query = "SELECT drivers.id, drivers.name, drivers.license_number "
                + " FROM cars INNER JOIN cars_drivers cs ON cs.car_id = cars.id "
                + "    INNER JOIN drivers ON cs.driver_id = drivers.id "
                + "WHERE cars.id = ? AND drivers.deleted = false ORDER BY drivers.id;";
```
what about if car was deleted?

Reply:
here you are fetching drivers for cars you have been selected in the previous query, so among them should not be deleted ones